packages/block-editor/src/components/link-control/index.js

```diff
@@ -133,14 +135,33 @@ function LinkControl( {
 		withCreateSuggestion = true;
 	}
 
+	const { settingsDrawerStatePreference } = useSelect( ( select ) => {
+		const prefsStore = select( preferencesStore );
+
+		const postEditorEnabled =
+			prefsStore.get( 'core/edit-post', 'linkControlSettingsDrawer' ) ??
+			false;
+
+		const siteEditorEnabled =
+			prefsStore.get( 'core/edit-site', 'linkControlSettingsDrawer' ) ??
+			false;
+
+		return {
+			settingsDrawerStatePreference:
+				postEditorEnabled || siteEditorEnabled,
+		};
```

Comment:
The return value should be determined based on which editor we are currently within. Else. setting from post editro will apply in Site Editor (and vice versa).

Any ideas or prior art on this. Maybe @talldan will know as he was the architect of the prefs store? 🙏